### ISSUE TITLE
The spartan movie...

### ISSUE THREAD

**Issue by yossigil** [SpartanRefactoring/Spartanizer#362]
Add a new handler and a new command, to do spartanization step by step. This means, find a tip, apply it, repeat. I think this would be fun!
What do you think @OriRoth

**Comment by yossigil** [comment]
Do this second... It is important since it would be a good selling point. But, it may take more than 10 minutes (not much more though)

**Comment by OriRoth** [comment]
The Spartan movie is out, and it is better than 300! (https://en.wikipedia.org/wiki/300_(film))
Use keyboard shortcut `Ctrl+Alt+M` to start the movie.
**Beware:** run time is estimated to be `[number of spartanization suggestions] * 1 + 2` seconds, so it is not recommended to run it on large projects.

**Comment by yossigil** [comment]
@orimarco , @M4tt30rru

**Comment by OriRoth** [comment]
![spartanization](https://cloud.githubusercontent.com/assets/15183108/19212649/59d65e3e-8d5e-11e6-9940-ac7a070be7d6.gif)

**Comment by OriRoth** [comment]
Now also on the readme page: https://github.com/SpartanRefactoring/Spartanizer

**Comment by yossigil** [comment]
:+1:

**Comment by OriRoth** [comment]
@yossigil I saw you are interested in improving the `SpartanMovie`. The current implementation works, but is *defiantly* non releasable - do you want me to make it more usable? (require some work)

**Comment by yossigil** [comment]
No. This is not a product, but I hope @OElmak  will do something about it.

**Comment by OriRoth** [comment]
@OElmak check your Telegram I have sent you a message. If you take this challenge you'll need some background story.

**Comment by yossigil** [comment]
@M4tt30rru We need your urgent help here.

**Comment by OriRoth** [comment]
Why? This is a pure GUI issue... threads and stuff.

**Comment by yossigil** [comment]
@M4tt30rru  has an implementation and a demo

**Comment by yossigil** [comment]
Matteo?

**Comment by OriRoth** [comment]
@yossigil you may consider giving it to the yearly project team...

**Comment by yossigil** [comment]
i want of off load you guys, as much as I can.

**Comment by M4tt30rru** [comment]
@yossigil I wrote to you on Telegram yesterday. @OriRoth I can start working on it today. I was abroad till yesterday.

**Comment by M4tt30rru** [comment]
The Spartan movie works on my computer, even if it does not show the progress (as in the video that @OriRoth posted on the main page). I tested it on a number of projects. I can start working on this. Are there any other improvements should I work on?

**Comment by yossigil** [comment]
Can you publish it?

**Comment by M4tt30rru** [comment]
I pushed the changes. It is now on the main menu.

**Comment by yossigil** [comment]
But isn't this the old version?

**Comment by M4tt30rru** [comment]
Exactly, this was actually my point. There is the problem I reported: it "freezes" and take some time to finish in some projects. I am working now on this "freezing" problem. For the rest it seems it is working fine.

**Comment by yossigil** [comment]
Great

**Comment by yossigil** [comment]
Done?

**Comment by M4tt30rru** [comment]
There are some progress, but I am still working on it.
